subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/plugins/PluginBuildsIntegrationTest.groovy

```diff
@@ -624,4 +625,114 @@ class PluginBuildsIntegrationTest extends AbstractPluginBuildIntegrationTest {
         where:
         dsl << ['Groovy', 'Kotlin']
     }
+
+    def "multiple included plugin builds which each include a library build with the same name works with #dsl dsl"() {
```

Comment:
I'm not sure if this is testing anything in particular.  The `lib`s are identical, so if we were getting the "wrong" one, you wouldn't notice.